codex-ui/src/vue/components/alert/useAlert.ts

```diff
@@ -0,0 +1,73 @@
+import type { Ref } from 'vue';
+import { ref } from 'vue';
+import { createSharedComposable } from '@vueuse/core';
+import type { AlertOptions, Alertype } from './Alert.types';
+
+export interface UseAlertComposableState {
+  /**
+   * Iterated store of alerts
+   */
+  alerts: Ref<AlertOptions[]>;
+
+  /**
+   * trigger success alert
+   * @param opt - alert options
+   */
+  success: (opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger error alert
+   * @param opt - alert options
+   */
+  error: (opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger warning alert
+   * @param opt - alert options
+   */
+  warning: (opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger info alert
+   * @param opt - alert options
+   */
+  info: (opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger default alert
+   * @param opt - alert options
+   */
+  alert: (opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+}
+
+/**
+ * Alert service hook
+ */
+export const useAlert = createSharedComposable((): UseAlertComposableState => {
+  const alerts = ref<AlertOptions[]>([]);
+
+  /**
+   * Trigger alert component
+   * @param type type of alert (success, error, warning, info and default)
+   * @param opt alert options(timeout, message and icon)
+   */
+  function triggerAlert(type?: Alertype, opt?: Pick<AlertOptions, 'id' | 'icon' | 'message' | 'timeout'>): void {
```

Comment:
as i can see, you are using `triggerAlert` only inside of this composable, so why should type and opt be optional?
i suggest making of the `type` and `opts` required, also `message` field should not be optional in `AlertOptions` to avoid empty alerts

Reply:
i think passing a default message should be more helpful, something like 

`message: 'codex' `over

`message: '' (empty string)`